### PR TITLE
Default cmap 'jet' -> 'viridis'

### LINKED ISSUE #55: Make default colormap consistent with matplotlib (jet -> viridis/coolwarm)
Several years ago matplotlib shifted from using the colormap [jet](https://www.mathworks.com/help/matlab/ref/jet.html) as its default, which was a legacy from when it imitated MATLAB.

jet is an extremely flawed and problematic colormap that should be avoided if possible (see below for details).

Matplotlib's change was precipitated by MATLAB shifting their default to [parula](https://uk.mathworks.com/help/matlab/ref/parula.html?searchHighlight=parula&s_tid=srchtitle), a more perceptually uniform colormap, for the above reasons. MATLAB copyrighted this colormap so matplotlib couldn't use it, as such they devised their own default and made it freely available to anyone, and everyone, who wishes to use it, [viridis](https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html).

Matplotlib have detailed their logic in changing away from jet and choosing viridis in a great talk I would recommend watching, which can be found [here](https://www.youtube.com/watch?v=xAoljeRJ3lU&ab_channel=Enthought). There is also a blog post to go along with it, which can be read [here](https://bids.github.io/colormap/).

# The issue

`jet` is not perceptually uniform; the rate of change of data is not proportional to the rate of change of lightness of color, in fact it is quite non-linear. In addition it is not color-blind friendly. `jet` distorts data. 

We should stop using jet as our default and shift to either viridis, (because viridis is the chosen default of matplotlib, bringing us in line with them), or to `coolwarm`. `coolwarm` should be considered because there has been research into which colormap looks best in 3D space, and divergent colormaps are generally considered superior. `coolwarm` is considered the best of the divergent colormaps (see [here](https://www.kennethmoreland.com/color-maps/) for details). 

`coolwarm` is the default in [Paraview](https://www.paraview.org/) for comparison.

## Online resources

- https://eagereyes.org/basics/rainbow-color-map
    - Blog post, with real example of how ‘rainbow’ misleads - Robert Kosara, Senior Research Scientist at Tableau Software and formerly Associate Professor of Computer Science at UNC Charlotte
 
- https://root.cern.ch/rainbow-color-map
    - Cern website post. Contains multiple real examples using Higgs-Boson data -Bernice Rogowitz (Visual Perspectives) and Olivier Couet (CERN).
    
- http://matplotlib.org/users/colormaps.html
    - Matplotlib has a good guide to cmaps on its website
    
- https://mycarta.wordpress.com/2012/05/29/the-rainbow-is-dead-long-live-the-rainbow-series-outline/
    - In depth, geophysics blog post series about why rainbow is bad, and what can be done
    
- http://www.scribblelive.com/blog/2011/12/07/the-use-of-yellow-in-data-design/
    - Short guide to why yellow is problematic in color schemes
    
- http://earthobservatory.nasa.gov/blogs/elegantfigures/2013/08/05/subtleties-of-color-part-1-of-6/
    - 6 part guide to colour usage in science

## Literature 

- Rainbow Color Map (Still) Considered Harmful’ in IEEE Computer Graphics and Applications, vol. 27, no. 2, pp. 14-17, March-April 2007.

- "Evaluation of Artery Visualizations for Heart Disease Diagnosis", Frank Rybicki, Simone Melchionna, Michelle Borkin, Hanspeter Pfister, Charles Feldman, Dimitrios Mitsouras, Krzysztof Gajos, Amanda Peters, IEEE Transactions on Visualization & Computer Graphics vol. 17 no. undefined, p. 2479-2488, Dec., 2011

- ‘Using color to code quantity in spatial displays.’ Spence, Ian; Kutlesa, Natasha; Rose, David L. Journal of Experimental Psychology: Applied, Vol 5(4), Dec 1999, 393-412

- Rainbow Color Map Critiques: An Overview and Annotated Bibliography By Steve Eddins, MathWorks *

- "Data visualization: the end of the rainbow," B. E. Rogowitz and L. A. Treinish, in IEEE Spectrum, vol. 35, no. 12, pp. 52-59, Dec 1998.

- Diverging color maps for scientific visualization - K. Moreland 2009, pp 92-103

- ColorBrewer.org: An Online Tool for Selecting Colour Schemes for Maps, The Cartographic Journal, 2003, pp 21-37

### PR DESCRIPTION
**Pull request by jgd10**
In the absence of any input on the topic in 6 months I have changed the
default to viridis in the commit. It is trivial to change to something
else if wanted, but that can be discussed in this (or a future) PR if anyone feels
strongly about it.

fixes #55

### CODE REVIEW

File: ansys/mapdl/reader/rst.py [hunk 1]
```diff
@@ -2741,7 +2741,7 @@ def _plot_point_scalars(self, scalars, rnum=None, grid=None,
             rng = kwargs.pop('rng', None)
 
         return_cpos = kwargs.pop('return_cpos', False)
-        cmap = kwargs.pop('cmap', 'jet')
+        cmap = kwargs.pop('cmap', 'viridis')
```
Comment (akaszynski) [inline ansys/mapdl/reader/rst.py:2744]: Let's provide the ability to override this and to use the theme parameters. If we hard code this, then it's impossible to change or override.

File: ansys/mapdl/reader/rst.py [hunk 2]
```diff
@@ -2961,7 +2961,7 @@ def _animate_point_scalars(self, scalars, grid=None,
             scalars[i] = scalars[i][ind]
 
         rng = kwargs.pop('rng', [np.min(scalars), np.max(scalars)])
-        cmap = kwargs.pop('cmap', 'jet')
+        cmap = kwargs.pop('cmap', 'viridis')
```
Comment (akaszynski) [inline ansys/mapdl/reader/rst.py:2964]: Same here.

File: ansys/mapdl/reader/rst.py (hunk 1 above)
Comment (jgd10) [reply ansys/mapdl/reader/rst.py:2744]: Do you mean we shouldn't hardcode a default?

Comment (akaszynski) [reply ansys/mapdl/reader/rst.py:2744]: We're good for now, we can have a follow-up that optimizes our use of themes.

**Review by akaszynski — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**